### PR TITLE
Rwa 844 - SearchForCompletable endpoint plugin

### PR DESCRIPTION
**Pull request by indvamshi**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-844

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by raghera — DISMISSED** [review]

**Review by dharmendrak — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerCFTTest.java [hunk 1]
```diff
@@ -0,0 +1,747 @@
...
+    private Headers authenticationHeaders;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Before
+    public void setUp() {
+        //Reset role assignments
+        authenticationHeaders = authorizationHeadersProvider.getTribunalCaseworkerAAuthorization("wa-ft-test-r2");
+        common.clearAllRoleAssignments(authenticationHeaders);
```
Comment (alexbaeza) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerCFTTest.java:72]: No need to clear all role assignments on the before each test will be using a single user and the roles are cleared as part of the setupOrganisationalRoleAssignment

**Review by alexbaeza — CHANGES_REQUESTED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerCFTTest.java [hunk 2]
```diff
@@ -0,0 +1,747 @@
...
+        );
+
+        // assign user to taskId3
+        final String taskId3 = tasksList.get(2).getId();
+        restApiActions.post(
+            "task/{task-id}/assign",
+            taskId3,
+            new AssignTaskRequest(assigneeId),
+            authenticationHeaders
+        );
```
[the remark this answers is not in the document]
Comment (indvamshi) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerCFTTest.java:415]: The above test is related to autoAssign so its ignored. I will delete the test as this tests are covered in another ticket.

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerCFTTest.java [hunk 3]
```diff
@@ -0,0 +1,747 @@
...
+                        result.then()
+                            .extract()
+                            .jsonPath().getList("", CamundaTask.class)
+                    );
+
+                    return true;
+                });
+
+        return response.get();
+    }
```
[the remark this answers is not in the document]
Comment (indvamshi) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerCFTTest.java:714]: yes, but without third argument, multiple FTs use the method so have to change in most FTs.

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/RoleAssignmentFilter.java [hunk 4]
```diff
@@ -256,4 +302,35 @@ private static Predicate searchByRoleLocation(Root<TaskResource> root,
...
+
+    private static Optional<RoleAssignment> filterByActiveRole(RoleAssignment roleAssignment) {
+        if (hasBeginTimePermission(roleAssignment) && hasEndTimePermission(roleAssignment)) {
+            return Optional.of(roleAssignment);
+        }
+        return Optional.empty();
+    }
+
+    private static boolean hasEndTimePermission(RoleAssignment roleAssignment) {
+        LocalDateTime endTime = roleAssignment.getEndTime();
```
[the remark this answers is not in the document]
Comment (indvamshi) [reply src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/RoleAssignmentFilter.java:314]: Its Provider-Consumer contract, they should tell us if they change to ZonedDateTime.

**Review by alexbaeza — APPROVED** [review]

**Pull request merged**